ykcompile/src/lib.rs

```diff
@@ -803,6 +803,148 @@ impl<TT> TraceCompiler<TT> {
         Ok(())
     }
 
+    fn c_binop(&mut self, binop: &BinOp, op1: &Operand, op2: &Operand) -> Location {
+        match binop {
+            BinOp::Eq | BinOp::Ne | BinOp::Lt | BinOp::Le | BinOp::Gt | BinOp::Ge => {
+                self.c_condition(binop, op1, op2)
+            }
+            _ => todo!(),
+        }
+    }
+
+    fn c_condition(&mut self, binop: &BinOp, op1: &Operand, op2: &Operand) -> Location {
+        let (src1, ty) = match op1 {
+            Operand::Place(p) => self.place_to_location(&p, false),
+            Operand::Constant(Constant::Int(_)) => todo!(),
+            Operand::Constant(Constant::Bool(_b)) => todo!(),
+            Operand::Constant(c) => todo!("{}", c),
+        };
+        let src2 = match op2 {
+            Operand::Place(p) => self.c_place(&p),
+            Operand::Constant(Constant::Int(ci)) => self.c_constint(&ci),
+            Operand::Constant(Constant::Bool(_b)) => todo!(),
+            Operand::Constant(c) => todo!("{}", c),
+        };
+
+        match (&src1, &src2) {
```

Comment:
All of these cases are a pain, but I can't think how to remove them all.

Reply:
We may be able to replace some of them by macros. But let's see how it looks like after the IR refactor.